ASIM/dev/ASimTester/filteringTest/filteringTest.py

```diff
@@ -0,0 +1,227 @@
+            empty_query = ""
+            response = client.query_workspace(
+                    workspace_id = ws_id, 
+                    query = empty_query,
+                    timespan = timedelta(days = 1)
+                    )
+            if response.status == LogsQueryStatus.PARTIAL or response.status == LogsQueryStatus.FAILURE:
+                raise Exception()
+            else:
+                return client
+    except Exception as e:
+        return None
+    
+
+# Authenticating the user
+client = attempt_to_connect(DefaultAzureCredential(exclude_interactive_browser_credential = False))
+if client is None:
+    client = attempt_to_connect(DefaultAzureCredential(exclude_interactive_browser_credential = False, exclude_shared_token_cache_credential = True))
+    if client is None:
+        print("Couldn't connect to workspace")
+        sys.exit()
+
+
+def get_parser(parser_path):
+    try:
+        with open(parser_path, 'r') as file_stream:
+            return yaml.safe_load(file_stream)
+    except:
+        raise Exception()
+
+
+# Creating a string of the parameters of a parser
+def create_parameters_string(parser_file):
+    paramsList = []
+    for param in parser_file['ParserParams']:
+        paramDefault = f"\'{param['Default']}\'" if param['Type']=="string" else param['Default']
+        paramsList.append(f"{param['Name']}:{param['Type']}={paramDefault}")
+    return ','.join(paramsList)
+    
+
+def create_query_without_call(parser_file):
+    params_str = create_parameters_string(parser_file) 
+    query_from_yaml = parser_file['ParserQuery']
+    return f"let query= ({params_str}) {{ {query_from_yaml} }};\n"
+
+
+def create_call_without_parameter(column_name):
+    return f"query() | summarize count() by {column_name}\n" 
+
+
+def create_call_with_parameter(parameter, value, column_name):
+    return f"query({parameter}={value}) | summarize count() by {column_name}\n"
+
+
+class FilteringTest(unittest.TestCase):
+    # "Main" function which runs initiates the validations 
+    def tests_main_func(self):
+        if not os.path.exists(parser_file_path):
+            self.fail(f"File path does not exist: {parser_file_path}")
+        if not parser_file_path.endswith('.yaml'): 
+            self.fail(f"Not a yaml file: {parser_file_path}")
+        try:
+            parser_file = get_parser(parser_file_path)
+        except:
+            self.fail(f"Cannot open file: {parser_file_path}")
+        self.check_required_fields(parser_file)
+        no_call_query = create_query_without_call(parser_file)
+        self.check_data_in_workspace(no_call_query)
+        columns_in_answer = self.get_columns_of_parser_answer(no_call_query)
+        if parser_file['Normalization']['Schema'] not in all_schemas_parameters:
+            self.fail(f"Schema: {parser_file['Normalization']['Schema']} - is not supported")
+        param_to_column_mapping = all_schemas_parameters[parser_file['Normalization']['Schema']]
```

Comment:
Is it enough? Maybe add more description to the error?

Reply:
Added more description.

Reply:
Please ask Ofer to rephrase it